src/Service/NotificationService.php

```diff
@@ -136,66 +128,4 @@ public function notificationHandler(array $data, array $configuration): array
         return $data;
 
     }//end notificationHandler()
-
-    /**
-     * Tries to find a synchronisation with the data from the notification and action->configuration.
-     *
-     * @throws Exception If we could not find a Source or Entity we throw an exception.
-     *
-     * @return Synchronization
-     */
-    private function findSync(): Synchronization
-    {
-        $dot = new Dot($this->data);
-        $url = $dot->get($this->configuration['urlLocation']);
-
-        // Find source by resource url from the notification.
-        $source = $this->findSource($url);
-
-        // Get the correct Entity.
-        $entity = $this->resourceService->getSchema($this->configuration['entity'], 'commongateway/corebundle');
-        if ($entity === null) {
-            throw new Exception("Could not find an Entity with this reference: {$this->configuration['entity']}", 500);
-        }
-
-        // Get (source) id from notification data.
-        $explodedUrl = explode('/', $url);
-        $sourceId    = end($explodedUrl);
-        if (isset($this->configuration['sourceIdUrl']) === true && $this->configuration['sourceIdUrl'] === true) {
-            $sourceId = $url;
-        }
-
-        $synchronization = $this->syncService->findSyncBySource($source, $entity, $sourceId);
-        $synchronization->setEndpoint(str_replace($source->getLocation(), '', $url));
-        $this->entityManager->persist($synchronization);
-
-        return $synchronization;
-
-    }//end findSync()
```

Comment:
Have you checked where this function is used before deleting it?

Reply:
Yess, it is a private function, only used in NotificationService